File: backend/src/main/java/com/streaker/controller/reward/dto/RewardResponseDto.java
package com.streaker.controller.reward.dto;

import java.time.Instant;
import java.util.UUID;

public record RewardResponseDto(
        UUID uuid,
        String name,
        String description,
        int pointsRequired,
        boolean unlocked,
        Instant unlockedAt,  // Nullable field
        UUID userId
) {}

File: backend/src/main/java/com/streaker/controller/reward/dto/RewardResponseDto.kt
package com.streaker.controller.reward.dto

import java.time.Instant
import java.util.UUID

data class RewardResponseDto(
    val uuid: UUID,
    val name: String,
    val description: String,
    val pointsRequired: Int,
    val unlocked: Boolean,
    var unlockedAt: Instant?,
    val userId: UUID
)
